ws/api/endpoints.go

```diff
@@ -16,25 +19,54 @@ import (
 	"github.com/go-chi/chi/v5"
 )
 
-var channelPartRegExp = regexp.MustCompile(`^\/?m\/([\w\-]+)\/c\/([\w\-]+)(\/[^?]*)?(\?.*)?$`)
+var (
+	channelPartRegExp = regexp.MustCompile(`^\/?m\/([\w\-]+)\/c\/([\w\-]+)(\/[^?]*)?(\?.*)?$`)
 
-func handshake(ctx context.Context, svc ws.Service) http.HandlerFunc {
+	errGenSessionID = errors.New("failed to generate session id")
+)
+
+func generateSessionID() (string, error) {
+	b := make([]byte, 32)
+	if _, err := rand.Read(b); err != nil {
+		return "", errors.Wrap(errGenSessionID, err)
+	}
+	return hex.EncodeToString(b), nil
+}
+
+func handshake(ctx context.Context, svc ws.Service, logger *slog.Logger) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		req, err := decodeRequest(r)
 		if err != nil {
 			encodeError(w, err)
 			return
 		}
-		conn, err := upgrader.Upgrade(w, r, nil)
+
+		sessionID, err := generateSessionID()
+		if err != nil {
+			logger.Warn(fmt.Sprintf("Failed to generate session id: %s", err.Error()))
+			http.Error(w, "", http.StatusInternalServerError)
+			return
+		}
+
+		responseHeader := http.Header{}
+		responseHeader.Set("X-SMQ-WS-Session-ID", sessionID)
```

Comment:
Why do we use this custom header?

Reply:
To identify the WS client.

If two ws client uses same thing id (client id), then if any one disconnect , another will not get message
So to avoid it,  we need unique id for each ws connection.